lib/src/fsmonitor.rs

```diff
@@ -166,10 +189,16 @@ pub mod watchman {
                 .resolve_root(working_copy_root)
                 .await
                 .map_err(Error::ResolveRootError)?;
-            Ok(Fsmonitor {
+
+            let monitor = Fsmonitor {
                 client,
                 resolved_root,
-            })
+            };
+
+            if set_up_trigger && !monitor.is_trigger_registered().await? {
```

Comment:
suggestion: Add a comment that not checking if the trigger is registered will cause Watchman to reevaluate the query.

Reply:
done